src/main/java/com/sap/sgs/phosphor/fosstars/data/github/NvdEntryMatcher.java

```diff
@@ -97,7 +105,10 @@ private boolean match(Configurations configurations) {
           continue;
         }
 
-        if (match(cpeMatch.getCpe22Uri()) || match(cpeMatch.getCpe23Uri())) {
+        if (StringUtils.isNotEmpty(cpeMatch.getCpe22Uri())
+            && match(productFrom22Uri(cpeMatch.getCpe22Uri()))
+            || (StringUtils.isNotEmpty(cpeMatch.getCpe23Uri())
+                && match(productFrom23Uri(cpeMatch.getCpe23Uri())))) {
```

Comment:
Could you please extract this complex check to a separate method (or a couple of methods)? If possible, please try to simplify it.